lib/adapters/test/testplane.ts

```diff
@@ -0,0 +1,54 @@
+import {TestplaneTestResultAdapter} from '../test-result/testplane';
+import {TestStatus, UNKNOWN_ATTEMPT} from '../../constants';
+
+import type {TestAdapter} from './';
+import type {Test, Suite} from 'testplane';
+import type {ReporterTestResult} from '../test-result';
+
+export class TestplaneTestAdapter implements TestAdapter {
+    private _test: Test;
+
+    static create<T extends TestplaneTestAdapter>(this: new (test: Test) => T, test: Test): T {
+        return new this(test);
+    }
+
+    constructor(test: Test) {
+        this._test = test;
+    }
+
+    get originalTest(): Test {
```

Comment:
Is it possible to get rid of this? It leaks implementation details (raw `TestplaneTest` out of abstraction layer `TestAdapter`)

Not critical

Reply:
I don't like this method either, but I don't find a better solution how to send original test to `getScreenshotPath` from testplane library - https://github.com/gemini-testing/testplane/blob/master/src/config/browser-config.ts#L11.